compiler/rustc_const_eval/src/util/type_name.rs

```diff
@@ -125,8 +132,7 @@ impl<'tcx> Printer<'tcx> for AbsolutePathPrinter<'tcx> {
         args: &[GenericArg<'tcx>],
     ) -> Result<(), PrintError> {
         print_prefix(self)?;
-        let args =
-            args.iter().cloned().filter(|arg| !matches!(arg.kind(), GenericArgKind::Lifetime(_)));
+        let args = args.iter().cloned();
```

Comment:
no need to clone `args` in the `if` anymore.

This is just `if !args.is_empty() { self.generic_delimiters(|cx| cx.comma_sep(args.iter())) }` by now, is it not?